tools/integration_tests/rapid_appends/appends_test.go

```diff
@@ -34,353 +33,252 @@ const (
+	_, err := readHandle.WriteAt([]byte(data), int64(len(s.fileContent)))
+	require.NoError(s.T(), err)
+
+	// Validate content is not yet visible.
+	contentBeforeClose, err := client.ReadObjectFromGCS(ctx, storageClient, path.Join(testDirName, s.fileName))
+	require.NoError(s.T(), err)
+	assert.Equal(s.T(), s.fileContent, string(contentBeforeClose))
+
+	// Close handle and validate final content.
+	readHandle.Close()
+	expectedContent := s.fileContent + data
+	contentAfterClose, err := client.ReadObjectFromGCS(ctx, storageClient, path.Join(testDirName, s.fileName))
+	require.NoError(s.T(), err)
+	assert.Equal(s.T(), expectedContent, string(contentAfterClose))
 }
 
-func (t *SingleMountAppendsSuite) TestKernelShouldSeeUpdatedSizeOnAppends() {
+func (s *AppendsTestSuite) TestKernelShouldSeeUpdatedSizeOnAppends() {
+	if s.cfg.isDualMount {
+		s.T().Skip("This test is designed for a single-mount configuration.")
+	}
 	const initialContent = "dummy content"
-	flags := []string{"--enable-rapid-appends=true", "--write-block-size-mb=1"}
-	log.Printf("Running test with flags: %v", flags)
 
 	testCases := []struct {
 		name        string
 		expireCache bool
 	}{
-		{
-			name:        "validStatCache",
-			expireCache: false,
-		},
-		{
-			name:        "expiredStatCache",
-			expireCache: true,
-		},
+		{"validStatCache", false},
+		{"expiredStatCache", true},
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.name, func() {
-			t.mountPrimaryMount(flags)
-			defer t.unmountPrimaryMount()
-
-			// Initially create an unfinalized object.
-			t.createUnfinalizedObject()
-			defer t.deleteUnfinalizedObject()
-
-			filePath := path.Join(t.primaryMount.testDirPath, t.fileName)
-
-			// Append to the unfinalized object and close the file handle.
-			appendFileHandle := operations.OpenFileInMode(t.T(), filePath, os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
-			n, err := appendFileHandle.Write([]byte(initialContent))
-			require.NoError(t.T(), err)
-			require.NotZero(t.T(), n)
+		s.Run(tc.name, func() {
+			s.createUnfinalizedObject()
+			defer s.deleteUnfinalizedObject()
+			filePath := path.Join(s.primaryMount.testDirPath, s.fileName)
+
+			// Append to the object and close the file handle.
+			appendFileHandle := operations.OpenFileInMode(s.T(), filePath, os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
+			_, err := appendFileHandle.Write([]byte(initialContent))
+			require.NoError(s.T(), err)
 			appendFileHandle.Close()
 
-			// Expire stat cache if required by the test case. By default, stat cache ttl is 1 sec.
 			if tc.expireCache {
 				time.Sleep(time.Second)
 			}
 
-			// stat() the file to assert on the file size as viewed by the kernel.
+			// Stat the file to assert on the file size as viewed by the kernel.
 			expectedFileSize := int64(unfinalizedObjectSize + len(initialContent))
 			fileInfo, err := operations.StatFile(filePath)
-			assert.NoError(t.T(), err)
-			assert.Equal(t.T(), expectedFileSize, (*fileInfo).Size())
+			assert.NoError(s.T(), err)
```

Comment:
If there is error, then checking the fileInfo.size() is meaningless.

```suggestion
			require.NoError(s.T(), err)
```